src/containerapp/azext_containerapp/_cloud_build_utils.py

```diff
@@ -0,0 +1,226 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+# pylint: disable=line-too-long
+
+from threading import Thread
+import os
+import re
+import requests
+import tempfile
+import time
+
+from ._archive_utils import archive_source_code
+
+from azure.cli.core.azclierror import (
+    ValidationError
+)
+
+from ._clients import BuilderClient, BuildClient
+
+from ._utils import (
+    log_in_file,
+    remove_ansi_characters
+)
+
+
+class CloudBuildError(Exception):
+    pass
+
+
+def run_cloud_build(cmd, source, location, resource_group_name, environment_name, run_full_id, logs_file, logs_file_path):
+    generated_build_name = 'build{}'.format(run_full_id)[:12]
+    log_in_file(f"Starting the Cloud Build for build of id '{generated_build_name}'\n", logs_file, no_print=True)
+
+    try:
+        done_spinner = False
+        fail_spinner = False
+        cancel_spinner = False
+        font_bold_green = "\033[32;1m"
+        font_bold_red = "\033[31;1m"
+        font_bold_yellow = "\033[33;1m"
+        font_default = "\033[0m"
+        font_bold = "\033[0;1m"
+        hide_cursor = "\033[?25l"
+        display_cursor = "\033[?25h"
+        substatus_indentation = "              "
+
+        def display_spinner(task_title):
+            # Hide the cursor
+            print(hide_cursor, end="")
+
+            def spin():
+                loop_counter = 0
+                start_time = time.time()
+                time_elapsed = 0
+                while done_spinner is False and fail_spinner is False and cancel_spinner is False:
+                    loop_counter = (loop_counter + 1) % 17
+                    loading_bar_left_spaces_count = loop_counter - 9 if loop_counter > 9 else 0
+                    loading_bar_right_spaces_count = 6 - loop_counter if loop_counter < 7 else 0
+                    spinner = f"[{' ' * loading_bar_left_spaces_count}{'=' * (7 - loading_bar_left_spaces_count - loading_bar_right_spaces_count)}{' ' * loading_bar_right_spaces_count}]"
+                    time_elapsed = time.time() - start_time
+                    print(f"\r    {spinner} {task_title} ({time_elapsed:.1f}s)", end="", flush=True)
```

Comment:
avoid use print(), see https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md#general-patterns.

is it possible to use progress bar in https://github.com/Azure/azure-cli/blob/main/src/azure-cli-core/azure/cli/core/commands/progress.py?

Reply:
This one is tricky. We've been asked to match (or as close as possible) to the Azure Dev CLI style, and I needed more freedom than what the typical Logger or Progress Bar gave me.
I agree that the style is different from what is usually done with the Az CLI, but in the end I think it is better for the customers.